examples/pennylane_run_doubly_stochastic.py

```diff
@@ -146,30 +151,30 @@ def circuit(params):
 opt = qml.GradientDescentOptimizer(eta)
 
 for _ in range(steps):
-    cost_GD.append(qnode_GD(params_GD))
-    params_GD = opt.step(qnode_GD, params_GD)
+    cost_GD.append(qnode_analytic(params_GD))
+    params_GD = opt.step(qnode_analytic, params_GD)
 
 # Optimizing using stochastic gradient descent with shots=1
 
-dev_SGD.shots = 1
+dev_stochastic.shots = 1
 cost_SGD1 = []
 params_SGD1 = init_params
 opt = qml.GradientDescentOptimizer(eta)
 
 for _ in range(steps):
-    cost_SGD1.append(qnode_SGD(params_SGD1))
-    params_SGD1 = opt.step(qnode_SGD, params_SGD1)
+    cost_SGD1.append(qnode_stochastic(params_SGD1))

```

Comment:
I would use `qnode_analytic` here, because otherwise you get a very weird cost graph that will totally confuse people. I would add an explanation why the cost is not evaluated on the same qnode though.